challenges/cloud/sre.md

```diff
@@ -0,0 +1,376 @@
+* Split the submission into 2-3 pull requests for us to review. We will review
+  every pull request and provide our feedback.
+* We are going to inspect the repository, test it, and get back to you.
+
+## Server
+
+An incomplete implementation of a Go server is located [here](sre-server.go).
+You will extend this implementation by adding features described below depending
+on the level you are applying to. Start by creating a new Github repository. Then
+let the interview panel know the repository's location by pasting a link in your
+interview Slack channel. Invite interview panel participants as contributors to
+the new repository if you prefer to keep your submission private.
+
+## Design Doc
+
+Start with a brief design document that covers the edge cases and design approach. At
+Teleport, we prefer Markdown for
+[our designs](https://github.com/gravitational/teleport/blob/master/rfd/0000-rfds.md).
+
+Be sure to cover the following in your design:
+
+* Developer workflow
+* Level 3+: Build, Release
+* Level 4+: State management and Delivery
+* Level 5+: Reconciliation, Conflicts, and Automation
+
+Open a pull request with your design and share a link with the reviewers via Slack.
+After the document is approved by at least 2 reviews, move on to the implementation.
+
+A few notes about the design document:
+
+* We expect the design document to be completed roughly within the first week.
+  This is to ensure you have enough time to work on the implementation.
+* Avoid writing an overly detailed design document. Two to three pages is
+  sufficient.
+* Avoid sending us draft design documents, it is difficult to evaluate what
+  parts are draft and which parts are complete. Instead we encourage asking
+  questions in Slack and sharing a design document that is ready to be
+  reviewed.
+
+## Server & Automation
+
+Once the design document is approved, begin working on adding features to the
+server implementation that includes automation. A key feature of this challenge 
+is to produce a self-contained GitHub project that automates away as much as 
+possible. This should include code quality, testing, build, and deployment.
+
+Expect the interview panel to clone the repository and execute one or more 
+Make targets that build and test a working solution. 
+
+Add a couple of high quality tests that cover happy and unhappy scenarios. Keep
+external dependencies as low or clear as possible. Expect the reviewers to
+checkout the repository and execute the tests themselves.
+
+Do not try to achieve full test coverage. This will take too long. Write enough
+to exercise the different key components and show they are working as
+intended, as well as to demonstrate your approach to automation.
+
+## Deployment
+
+Levels 3+ include a deployment step which should deploy the solution to a local
+Kubernetes cluster. The choice of which Kubernetes cluster is up to you, but 
```

Comment:
> The choice of which Kubernetes cluster is up to you

This appears to conflict with later guidance to use AWS EKS.  If the intention is to have two deploy targets (e.g. minikube/kind _and_ EKS, that could be clarified).

Reply:
We are only asking that they include an EKS section in the design.